cmd/flux/create_secret_git.go

```diff
@@ -147,8 +149,9 @@ func createSecretGitCmdRun(cmd *cobra.Command, args []string) error {
 		opts.ECDSACurve = secretGitArgs.ecdsaCurve.Curve
 		opts.Password = secretGitArgs.password
 	case "http", "https":
-		if secretGitArgs.username == "" || secretGitArgs.password == "" {
-			return fmt.Errorf("for Git over HTTP/S the username and password are required")
+		if secretGitArgs.username != "" || secretGitArgs.password != "" {
+			opts.Username = secretGitArgs.username
+			opts.Password = secretGitArgs.password
```

Comment:
why are we removing the check for empty username/password here? this should remain and we should add a check for bearer tokens as well. something like:
```go
if secretGitArgs.username == "" || secretGitArgs.password == "" {
    if secretGitArgs.bearerToken == "" {
        return fmt.Errorf("for Git over HTTP/S the username and password are required")
    }
}
```